rust/twitch_utils/src/twitchtypes.rs

```
@@ -65,10 +65,21 @@ impl From<ChannelEmote> for TwitchEmote {
 
 impl From<RawSevenTVEmote> for SevenTVEmote {
     fn from(raw_emote: RawSevenTVEmote) -> Self {
-        Self {
-            id: raw_emote.id,
-            name: raw_emote.name,
-            emote_url: raw_emote.host.url,
+        let largest_width_file = raw_emote.host.files.iter().max_by_key(|file| file.width);
+        if let Some(file) = largest_width_file {
+            let url = raw_emote.host.url + "/" + &file.name;
+            Self {
+                id: raw_emote.id,
+                name: raw_emote.name,
+                emote_url: url,
+            }
+        } else {
+            // Use technical difficulties emote if no files are found
+            Self {
+                id: raw_emote.id,
+                name: raw_emote.name,
+                emote_url: String::from("https://cdn.7tv.app/emote/63384017cf7eb48c4e731a79/4x.webp"),
+            }
```

qns: were there any cases where this is needed?

---

I've never seen it happen. Do you think we should throw an error if this happens?

---

Why doesn't the old method work?

---

The old method grabbed the root url for the emote, which can't be used to directly display the emote. This final step of adding the specific image was elsewhere before this branch was split for PRs, but now I think it's most appropriate here.

---

Oh, so was the old PR not fully functional, then? Seems like something that escaped my review again, lol...